src/main/java/org/folio/orders/utils/HelperUtils.java

```diff
@@ -393,59 +386,16 @@ public static CompositePurchaseOrder convertToCompositePurchaseOrder(JsonObject
     return poJson.mapTo(CompositePurchaseOrder.class);
   }
 
-  public static boolean changeOrderStatus(PurchaseOrder purchaseOrder, List<PoLine> poLines) {
-    if (toBeCancelled(purchaseOrder, poLines)) {
-      purchaseOrder.setWorkflowStatus(PurchaseOrder.WorkflowStatus.CLOSED);
-      purchaseOrder.setCloseReason(new CloseReason().withReason(REASON_CANCELLED));
-      return true;
-    }
-
-    if (toBeClosed(purchaseOrder, poLines)) {
-      purchaseOrder.setWorkflowStatus(PurchaseOrder.WorkflowStatus.CLOSED);
-      purchaseOrder.setCloseReason(new CloseReason().withReason(REASON_COMPLETE));
-      return true;
-    }
-
-    if (toBeReopened(purchaseOrder, poLines)) {
-      purchaseOrder.setWorkflowStatus(PurchaseOrder.WorkflowStatus.OPEN);
-      return true;
-    }
-
-    return false;
-  }
-
   public static String convertTagListToCqlQuery(Collection<String> values, String fieldName, boolean strictMatch) {
-
     String prefix = fieldName + (strictMatch ? "==(\"" : "=(\"");
     return StreamEx.of(values).joining("\" or \"", prefix, "\")");
   }
 
-  private static boolean toBeClosed(PurchaseOrder purchaseOrder, List<PoLine> poLines) {
-    return purchaseOrder.getWorkflowStatus() == PurchaseOrder.WorkflowStatus.OPEN
-      && poLines.stream().allMatch(HelperUtils::isCompletedPoLine);
-  }
-
-  private static boolean toBeCancelled(PurchaseOrder purchaseOrder, List<PoLine> poLines) {
-    return purchaseOrder.getWorkflowStatus() == PurchaseOrder.WorkflowStatus.OPEN
-      && poLines.stream().allMatch(HelperUtils::isCancelledPoLine);
-  }
-
-  private static boolean isCancelledPoLine(PoLine line) {
-    PoLine.PaymentStatus paymentStatus = line.getPaymentStatus();
-    PoLine.ReceiptStatus receiptStatus = line.getReceiptStatus();
-    return paymentStatus == PaymentStatus.CANCELLED && receiptStatus == ReceiptStatus.CANCELLED;
-  }
-
-  private static boolean toBeReopened(PurchaseOrder purchaseOrder, List<PoLine> poLines) {
-    return purchaseOrder.getWorkflowStatus() == PurchaseOrder.WorkflowStatus.CLOSED
-      && poLines.stream().anyMatch(line -> !isCompletedPoLine(line));
-  }
-
-  private static boolean isCompletedPoLine(PoLine line) {
-    PoLine.PaymentStatus paymentStatus = line.getPaymentStatus();
-    PoLine.ReceiptStatus receiptStatus = line.getReceiptStatus();
-    return (paymentStatus == PAYMENT_NOT_REQUIRED || paymentStatus == FULLY_PAID || paymentStatus == PaymentStatus.CANCELLED)
-      && (receiptStatus == FULLY_RECEIVED || receiptStatus == RECEIPT_NOT_REQUIRED || receiptStatus == ReceiptStatus.CANCELLED);
+  public static CompositePoLine convertToCompositePoLine(PoLine poLine) {
```

Comment:
we already have this method you can use it and modify if need. https://github.com/folio-org/mod-orders/blob/7bd52b48dbab62b64e1742eabb5609dd95c46b8d/src/main/java/org/folio/orders/utils/PoLineCommonUtil.java#L253

Reply:
I see this one modifies the poLine itself and used in many places, probably having alternate one that does not modify original object could be sensible, how do you think?

Reply:
Didn't notice, It is strange behaviour if one method will modify actual object and other don't. I recommend to keep only one and lets not modify actual object, so move your logic to existing one and remove current one.

Reply:
Could there be some dependency for removing those values? We could keep the old one and use new one in the future

Reply:
not sure, but I assume, changing old one to new one should affect to anywhere, since it is converting method. if it is, there is big problem old method. from my opinion, this thing only two of us know about situation but no one know when using these methods, leading to using old one because of located in PoLineCommonUtil. However, if you still want to keep both, please add some java doc to explain about method (not recommended from my side).

Reply:
Sure I see your point. Best would be to remove it and I will check if doing it will affect anything